Mods/interface/PuppiMod.h

```diff
@@ -57,17 +53,26 @@ namespace mithep
       void SetDump( Bool_t dump )                            { fDumpingPuppi = dump;           }
 
     protected:
-      void                  SlaveBegin();
-      void                  SlaveTerminate();
-      void                  Process();
+      void                  SlaveBegin() override;
+      void                  SlaveTerminate() override;
+      void                  Process() override;
+
+      enum ParticleType {
+        kChargedPrimary = 1,
+        kChargedPU,
+        kNeutralCentral,
+        kNeutralForward
+      };
+
+      ParticleType GetParticleType( const PFCandidate *cand, Vertex const* ) const;
+      Int_t GetEtaBin( const PFCandidate *cand ) const;
+      Double_t Chi2fromDZ( Double_t dz ) const;
       
       TString               fEtaConfigName;          // Name of the configuration file with eta tables
       TString               fVertexesName;           // Name of vertices collection used for PV
       TString               fPFCandidatesName;       // Name of PFCandidate collection (input)
       TString               fPuppiParticlesName;     // Name of Puppi Particle collection (output)
 
-      const VertexCol      *fVertexes;               // Vertex branch
```

Comment:
fVertexes was used only in Process() and GetParticleType, but the usage in GetParticleType was limited and could easily be removed by making the PV an argument of the function. With this update the vertex collection only resides in Process() and therefore does not have to be a class member variable any more.
The fewer the member variable is, the easier to understand what the class does.